screens/overviewMap.js

```diff
@@ -182,30 +218,72 @@ class OverviewMap extends PureComponent {
     }
   };
 
-  onDragStart = (event) => {
+  onTempPinDragStart = (event) => {
     this.setState({ movingTemp: true });
     event.persist();
   };
 
-  onDrag = (event) => event.persist();
+  onTempPinDrag = (event) => event.persist();
 
-  onDragEnd = (event) => {
+  onTempPinDragEnd = (event) => {
     const coord = event.nativeEvent.coordinate;
     this.setState({ movingTemp: false, tempLocation: coord });
     event.persist();
   };
 
-  onLocationCancel = () => {
-    this.setState({ tempLocation: null });
+  onPinDragStart(event, isUploaded) {
+    this.setState({ draggingPin: true, canDeletePin: !isUploaded });
+    event.persist();
+  }
+
+  onPinDrag(event) {
+    const { canDeletePin } = this.state;
+    const { position } = event.nativeEvent;
+    event.persist();
+
+    if (position.y < this.screenHeight / 8) {
+      if (position.x < this.screenWidth / 2 || !canDeletePin) {
+        this.setState({ inEdit: true, inDelete: false });
+        return;
+      }
+      this.setState({ inEdit: false, inDelete: true });
+      return;
+    }
+
+    this.setState({ inEdit: false, inDelete: false });
+  }
+
+  onPinDragEnd(event, pinLocation) {
+    const { inEdit, inDelete } = this.state;
+    const { deleteLocalLocation } = this.props;
+
+    if (inEdit) {
+      this.createTempPin({ latitude: pinLocation.latitude, longitude: pinLocation.longitude }, pinLocation.key);
+    } else if (inDelete) {
+      deleteLocalLocation(pinLocation.key);
+    } else {
+      // Is there a better way to get the pin to rerender??? Cause this is actually gross...
+      this.setState(() => ({ editingPinKey: pinLocation.key }), () => this.setState(() => ({ editingPinKey: null })));
```

Comment:
I think you might be able to call a method to force it to rerender. You are right. This is gross.

Reply:
Apparently, there is something weird with markers and they only fully redraw when their `key` prop changes so I had to use a `Date.now()` to build a unique key every time the `LocationMarker` component rerenders. But I think that's still better than the old way.

Reply:
Oh interesting? Are they pure or have a `shouldComponentUpdate`?

Reply:
As far as I can see it's not pure and doesn't have a `shouldComponentUpdate`

Reply:
Odd. Must be something from the maps library. I think I saw one of the items higher in the hierarchy was pure, but that could just be a red herring.

Reply:
Mind adding a comment as to why we are doing this odd things?